x-pack/test/performance/tests/playwright/flight_dashboard.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import Url from 'url';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function flightDashboard({ getService }: FtrProviderContext) {
+  describe('Flights Dashboard', () => {
+    const config = getService('config');
+    const playwright = getService('playwright');
+    const logger = getService('log');
+    const { step } = playwright.makePage({ autoLogin: true, journeyName: 'flights_dashboard' });
+
+    step('Go to Sample Data Page', async ({ page }) => {
+      const kibanaUrl = Url.format({
+        protocol: config.get('servers.kibana.protocol'),
+        hostname: config.get('servers.kibana.hostname'),
+        port: config.get('servers.kibana.port'),
+      });
+
+      await page.goto(`${kibanaUrl}/app/home#/tutorial_directory/sampleData`);
+      await page.waitForSelector('text="More ways to add data"');
+    });
+
+    step('Add Flights Sample Data', async ({ page }) => {
+      const removeButton = page.locator('[data-test-subj=removeSampleDataSetflights]');
+      try {
+        await removeButton.click({ timeout: 1_000 });
+      } catch (e) {
+        logger.info('Flights data does not exist');
+      }
+
+      const addDataButton = page.locator('[data-test-subj=addSampleDataSetflights]');
+      if (addDataButton) {
+        await addDataButton.click();
+      }
+    });
+
+    step('Go to Flights Dashboard', async ({ page }) => {
+      const viewdataBtn = page.locator('[aria-label="View Sample flight data"]');
```

Comment:
I usually try to use test explicit attributes but in this case it was:
```
<button class="euiButton euiButton--primary" type="button" aria-label="View Sample flight data">
```
so there wasn't any test-subj attribute and I didn't wanna use complicated selector like `[data-test-subj=launchSampleDataSetflights]>div>button`. But I can change it something like that If you see some benefit.

And for text locators I thought it is better idea and easier to manage to just use text visible on the screen.

Reply:
Right, I think we should add a `data-test-subj` attribute to that button if we're going to be using it for testing. Visible text on the screen and invisible text for a11y changes all the time and shouldn't break the functional tests.

Reply:
I agree and I was actually thinking more internalization can be issue. But I will add the attribute to the button in this case 👍🏽

Reply:
https://github.com/elastic/kibana/blob/9c3c32378096af685b766a44ed01bd0a8f223d35/src/plugins/home/public/application/components/sample_data_view_data_button.js#L52-L62

This component will generate this html structure:
<img width="1184" alt="Screenshot 2022-03-04 at 01 15 20" src="https://user-images.githubusercontent.com/9674241/156674812-5c9059c5-9b16-4ff1-9371-af813fca3b1f.png">

and in order to change the buttons attribute I have to change how _EuiButton_ passes the prop to inner button node. I'm not sure if we want to block this PR because of that rather we can have temporarily `[data-test-subj=launchSampleDataSetflights]>div>button` solution and then change later.

Reply:
We can't just click on `[data-test-subj=launchSampleDataSetflights]`? In Selenium I guess that works because it finds that element and then clicks in that area, it doesn't deliver a click event exactly, it just simulates a click, does that not work in playwright?

Reply:
That might work I tried by just trying document.querySelector().click() and as expected it didn't work.

===========================================================================

I now tried but unfortunately non of the playwright api's clicks the button, I used these in different runs: 
- [Page#click](https://playwright.dev/docs/api/class-page#page-click)
- [ElementHandle#click](https://playwright.dev/docs/api/class-elementhandle#element-handle-click)
- [Locator Api](https://playwright.dev/docs/locators) 
  - Without force (checks [actionability](https://playwright.dev/docs/actionability) of the element)
  - With force (bypass the actionability checks)

Reply:
That's sad to hear, I guess `[data-test-subj=launchSampleDataSetflights] button` would be the next best thing

Reply:
I changed all [text=".."] parts that I can change based on the html attributes. The ones I left is not selectable by any easy selector. `[data-test-subj=launchSampleDataSetflights] button` this also doesn't work `[data-test-subj=launchSampleDataSetflights]>div>button` is the only way if we want to use the **data-test-subj** 😔

Reply:
Huh? Why doesn't a child selector for `button` work? Sounds like something is going on that we need to investigate if basic CSS selectors don't work...